pkg/client/tx/client.go

```diff
@@ -404,6 +376,36 @@ func (tClient *txClient) goSubscribeToOwnTxs(ctx context.Context) {
 			if len(txErrorChans) == 0 {
 				delete(tClient.txTimeoutPool, timeoutHeight)
 			}
+
+			// Check for a corresponding error channel in the map.
+			// Only close and remove from tClient.txErrorChans if it has been initiated
+			// by tClient.
+			// Transactions that are initiated externally will not have an associated
```

Comment:
I don't fully understand why being inside the loop solves it. You explained the `what` but not the `why`

#PUC